src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -0,0 +1,191 @@
+  }
+
+  override def copy(extra: ParamMap): Transformer = defaultCopy(extra)
+}
+
+object TextAnalyticsLanguageDetection extends ComplexParamsReadable[TextAnalyticsLanguageDetection]
+
+class TextAnalyticsLanguageDetection(override val textAnalyticsOptions: Option[TextAnalyticsRequestOptions] = None,
+                                     override val uid: String = randomUID("TextAnalyticsLanguageDetection"))
+  extends TextAnalyticsSDKBase[DetectedLanguageV4](textAnalyticsOptions)
+  with HasConfidenceScoreCol {
+  logClass()
+
+  /**
+   * Params for optional input column names.
+   */
+  // Country Hint
+  final val countryHintCol: Param[String] = new Param[String](this, "countryHintCol", "country hint column name")
+
+  final def getCountryHintCol: String = $(countryHintCol)
+
+  final def setCountryHintCol(value: String): this.type = set(countryHintCol, value)
+  setDefault(countryHintCol -> "CountryHint")
+
+  override def outputSchema: StructType = DetectLanguageResponseV4.schema
+
+  override protected val invokeTextAnalytics: String => TAResponseV4[DetectedLanguageV4] = (text: String) =>
+    {
+      val detectLanguageResultCollection = textAnalyticsClient.detectLanguageBatch(
+        Seq(text).asJava, null, textAnalyticsOptions.orNull)
+      val detectLanguageResult = detectLanguageResultCollection.asScala.head
+
+      val languageResult = if (detectLanguageResult.isError) {
+        None
+      } else {
+        Some(DetectedLanguageV4(
+          detectLanguageResult.getPrimaryLanguage.getName,
+          detectLanguageResult.getPrimaryLanguage.getIso6391Name,
+          detectLanguageResult.getPrimaryLanguage.getConfidenceScore))
+      }
+
+      val error = if (detectLanguageResult.isError) {
+        val error = detectLanguageResult.getError
+        Some(TAErrorV4(error.getErrorCode.toString, error.getMessage, error.getTarget))
+      } else {
+        None
+      }
+
+      val stats = Option(detectLanguageResult.getStatistics) match {
+        case Some(s) => Some(DocumentStatistics(s.getCharacterCount, s.getTransactionCount))
+        case None => None
+      }
+
+      TAResponseV4[DetectedLanguageV4](
+        languageResult,
+        error,
+        stats,
+        Some(detectLanguageResultCollection.getModelVersion))
+    }
+}
+
+object DetectLanguageResponseV4 extends SparkBindings[TAResponseV4[DetectedLanguageV4]]
+object KeyPhraseResponseV4 extends SparkBindings[TAResponseV4[KeyphraseV4]]
+
+case class TAResponseV4[T](result: Option[T],
+                           error: Option[TAErrorV4],
+                           statistics: Option[DocumentStatistics],
+                           modelVersion: Option[String])
+
+case class TAErrorV4(errorCode: String, errorMessage: String, target: String)
+
+case class DetectedLanguageV4(name: String, iso6391Name: String, confidenceScore: Double)
+
+object TextAnalyticsKeyphraseExtraction extends ComplexParamsReadable[TextAnalyticsKeyphraseExtraction]

```

Comment:
Could we move the data objects and case classes into a separate file, like it is in the existing connector?